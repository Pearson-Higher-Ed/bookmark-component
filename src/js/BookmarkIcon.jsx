import React, { PropTypes } from 'react';
import { messages } from './defaultMessages';
import SvgIcon from 'material-ui/SvgIcon';

class BookmarkIcon extends React.Component {
  constructor(props) {
    super(props);
    /*const _isBookmarked = this.props.data.isCurrentPageBookmarked();
    this.state ={
      'isBookmarked': _isBookmarked
    };*/
  }

  /*componentDidMount() {
    this.pubsub_token = window.pubsub.subscribe('IS_BOOKMARKED', function(topic, isBookmarked) {
      if (isBookmarked) {
        this.setState({isBookmarked: true});
      } else {
        this.setState({isBookmarked: false});
      }
    }.bind(this));

  }

  componentWillUnmount() {
    window.pubsub.unsubscribe(this.pubsub_token);
  }
*/
  toggleBookmark() {

    const that = this;
    function removedBookmarkCbk() {
      that.setState({isBookmarked: false});
    }

    function addedBookmarkCbk() {
      that.setState({isBookmarked: true});
    }

    if (this.state.isBookmarked) {
      this.props.data.removeBookmarkHandler(removedBookmarkCbk);
    } 
    else {
      this.props.data.addBookmarkHandler(addedBookmarkCbk);
    }
  }

  render() {
    return (
  <SvgIcon >
    <path d="M11.5,-0.0004 L0.5,-0.0004 C0.224,-0.0004 0,0.2236 0,0.4996 L0,16.4996 C0,16.6996 0.119,16.8806 0.302,16.9586 C0.488,17.0396 0.699,16.9996 0.844,16.8636 L6,11.9876 L11.156,16.8636 C11.251,16.9526 11.375,16.9996 11.5,16.9996 C11.566,16.9996 11.634,16.9866 11.698,16.9586 C11.881,16.8806 12,16.6996 12,16.4996 L12,0.4996 C12,0.2236 11.776,-0.0004 11.5,-0.0004 L11.5,-0.0004 Z" fill="#FFF"></path>
  </SvgIcon>
    )
  }
}
BookmarkIcon.propTypes = {
 locale: PropTypes.string,
 data: PropTypes.shape({
   addBookmarkHandler: PropTypes.func,
   removeBookmarkHandler: PropTypes.func,
   isCurrentPageBookmarked: PropTypes.func
 })
};



export default BookmarkIcon;
