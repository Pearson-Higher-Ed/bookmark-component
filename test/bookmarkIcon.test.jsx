/**
PEARSON PROPRIETARY AND CONFIDENTIAL INFORMATION SUBJECT TO NDA
 *  Copyright Â© 2017 Pearson Education, Inc.
 *  All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of Pearson Education, Inc.  The intellectual and technical concepts contained
 * herein are proprietary to Pearson Education, Inc. and may be covered by U.S. and Foreign Patents,
 * patent applications, and are protected by trade secret or copyright law.
 * Dissemination of this information, reproduction of this material, and copying or distribution of this software
 * is strictly forbidden unless prior written permission is obtained
 * from Pearson Education, Inc.
**/
/* eslint-disable */

import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { IntlProvider } from 'react-intl';
import { InternationalSupport } from '@pearson-incubator/aquila-js-core';
import SvgIcon from 'material-ui/SvgIcon';
import BookmarkIcon from '../src/js/BookmarkIcon';
import msgObject from '../translations';

const isCurrentPageBookmarked = {
  isCurrentPageBookmarked: function() {
    return true
  },
  removeBookmarkHandler: function() {
    console.log('removeBookmarkHandler');
  }
}

const intlObj = new InternationalSupport(msgObject, 'en');
it('renders the component correctly', () => {
  const tree = renderer
    .create(
      <MuiThemeProvider>
        <IntlProvider 
        locale={intlObj.getLocale()} 
        messages={intlObj.getMessages()}>
          <BookmarkIcon data={isCurrentPageBookmarked}/>
        </IntlProvider>
      </MuiThemeProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render onClick', () => {
  const component = renderer.create(<MuiThemeProvider><IntlProvider 
      locale={intlObj.getLocale()} 
      messages={intlObj.getMessages()}><BookmarkIcon data={isCurrentPageBookmarked} /></IntlProvider></MuiThemeProvider>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
   // manually trigger the callback
  tree.children[0].children[0].props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
it('should render onKeyDown', () => {
  const component = renderer.create(<MuiThemeProvider><IntlProvider 
      locale={intlObj.getLocale()} 
      messages={intlObj.getMessages()}><BookmarkIcon data={isCurrentPageBookmarked} /></IntlProvider></MuiThemeProvider>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
   // manually trigger the callback
  tree.children[0].children[0].props.onKeyDown();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
it('should render onMouseEnter', () => {
  const component = renderer.create(<MuiThemeProvider><IntlProvider 
      locale={intlObj.getLocale()} 
      messages={intlObj.getMessages()}><BookmarkIcon data={isCurrentPageBookmarked} /></IntlProvider></MuiThemeProvider>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
   // manually trigger the callback
  tree.children[0].children[0].props.onMouseEnter();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
it('should render onMouseLeave', () => {
  const component = renderer.create(<MuiThemeProvider><IntlProvider 
      locale={intlObj.getLocale()} 
      messages={intlObj.getMessages()}><BookmarkIcon data={isCurrentPageBookmarked} /></IntlProvider></MuiThemeProvider>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
   // manually trigger the callback
  tree.children[0].children[0].props.onMouseLeave();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render SvgIcon', () => {
  const component = renderer.create(<MuiThemeProvider><IntlProvider 
      locale={intlObj.getLocale()} 
      messages={intlObj.getMessages()}><BookmarkIcon data={isCurrentPageBookmarked}><SvgIcon /></BookmarkIcon></IntlProvider></MuiThemeProvider>);
  expect(component.toJSON()).toMatchSnapshot();
});








