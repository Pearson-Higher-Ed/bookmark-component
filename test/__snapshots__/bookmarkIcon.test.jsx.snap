// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly 1`] = `
<div>
  <div
    aria-label="Bookmarked"
    className="filled"
  >
    <svg
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fill": "currentColor",
          "height": "20px",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "width": "16px",
        }
      }
      tabIndex="0"
      viewBox="0 0 15 18"
    >
      <path
        d="M11.5,-0.0004 L0.5,-0.0004 C0.224,-0.0004 0,0.2236 0,0.4996 L0,16.4996 C0,16.6996 0.119,16.8806 0.302, 16.9586 C0.488,17.0396 0.699,16.9996 0.844,16.8636 L6,11.9876 L11.156,16.8636 C11.251,16.9526 11.375, 16.9996 11.5,16.9996 C11.566,16.9996 11.634,16.9866 11.698,16.9586 C11.881,16.8806 12,16.6996 12, 16.4996 L12,0.4996 C12,0.2236 11.776,-0.0004 11.5,-0.0004 L11.5,-0.0004 Z"
        fill="#FFF"
      />
    </svg>
  </div>
</div>
`;

exports[`should render SvgIcon 1`] = `
<div>
  <div
    aria-label="Bookmarked"
    className="filled"
  >
    <svg
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fill": "currentColor",
          "height": "20px",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "width": "16px",
        }
      }
      tabIndex="0"
      viewBox="0 0 15 18"
    >
      <path
        d="M11.5,-0.0004 L0.5,-0.0004 C0.224,-0.0004 0,0.2236 0,0.4996 L0,16.4996 C0,16.6996 0.119,16.8806 0.302, 16.9586 C0.488,17.0396 0.699,16.9996 0.844,16.8636 L6,11.9876 L11.156,16.8636 C11.251,16.9526 11.375, 16.9996 11.5,16.9996 C11.566,16.9996 11.634,16.9866 11.698,16.9586 C11.881,16.8806 12,16.6996 12, 16.4996 L12,0.4996 C12,0.2236 11.776,-0.0004 11.5,-0.0004 L11.5,-0.0004 Z"
        fill="#FFF"
      />
    </svg>
  </div>
</div>
`;

exports[`should render onClick 1`] = `
<div>
  <div
    aria-label="Bookmarked"
    className="filled"
  >
    <svg
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fill": "currentColor",
          "height": "20px",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "width": "16px",
        }
      }
      tabIndex="0"
      viewBox="0 0 15 18"
    >
      <path
        d="M11.5,-0.0004 L0.5,-0.0004 C0.224,-0.0004 0,0.2236 0,0.4996 L0,16.4996 C0,16.6996 0.119,16.8806 0.302, 16.9586 C0.488,17.0396 0.699,16.9996 0.844,16.8636 L6,11.9876 L11.156,16.8636 C11.251,16.9526 11.375, 16.9996 11.5,16.9996 C11.566,16.9996 11.634,16.9866 11.698,16.9586 C11.881,16.8806 12,16.6996 12, 16.4996 L12,0.4996 C12,0.2236 11.776,-0.0004 11.5,-0.0004 L11.5,-0.0004 Z"
        fill="#FFF"
      />
    </svg>
  </div>
</div>
`;

exports[`should render onClick 2`] = `
<div>
  <div
    aria-label="Not Bookmarked"
    className="unfilled"
  >
    <svg
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fill": "currentColor",
          "height": "20px",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "width": "16px",
        }
      }
      tabIndex="0"
      viewBox="0 0 15 18"
    >
      <path
        d="M13.2266667,0 L0.48,0 C0.32,0 0.16,0.154285714 0.16,0.308571429 L0.16,17.2285714 C0.16,17.3314286 0.213333333, 17.4342857 0.32,17.4857143 C0.373333333,17.4857143 0.426666667,17.5371429 0.48,17.5371429 C0.533333333, 17.5371429 0.586666667,17.5371429 0.64,17.4857143 L6.72,13.5257143 L13.0666667,17.4857143 C13.12, 17.5371429 13.1733333,17.5371429 13.2266667,17.5371429 C13.28,17.5371429 13.3333333,17.5371429 13.3866667, 17.4857143 C13.4933333,17.4342857 13.5466667,17.3314286 13.5466667,17.2285714 L13.5466667,0.308571429 C13.4933333, 0.154285714 13.3866667,0 13.2266667,0 L13.2266667,0 Z M12.32,15.5828571 L7.41333333,12.4971429 L6.72, 12.0857143 L6.08,12.4971429 L1.44,15.5314286 L1.44,1.13142857 L12.3733333,1.13142857 L12.3733333, 15.5828571 L12.32,15.5828571 Z"
        fill="#FFF"
        id="Shape"
      />
    </svg>
  </div>
</div>
`;

exports[`should render onKeyDown 1`] = `
<div>
  <div
    aria-label="Bookmarked"
    className="filled"
  >
    <svg
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fill": "currentColor",
          "height": "20px",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "width": "16px",
        }
      }
      tabIndex="0"
      viewBox="0 0 15 18"
    >
      <path
        d="M11.5,-0.0004 L0.5,-0.0004 C0.224,-0.0004 0,0.2236 0,0.4996 L0,16.4996 C0,16.6996 0.119,16.8806 0.302, 16.9586 C0.488,17.0396 0.699,16.9996 0.844,16.8636 L6,11.9876 L11.156,16.8636 C11.251,16.9526 11.375, 16.9996 11.5,16.9996 C11.566,16.9996 11.634,16.9866 11.698,16.9586 C11.881,16.8806 12,16.6996 12, 16.4996 L12,0.4996 C12,0.2236 11.776,-0.0004 11.5,-0.0004 L11.5,-0.0004 Z"
        fill="#FFF"
      />
    </svg>
  </div>
</div>
`;

exports[`should render onMouseEnter 1`] = `
<div>
  <div
    aria-label="Bookmarked"
    className="filled"
  >
    <svg
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fill": "currentColor",
          "height": "20px",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "width": "16px",
        }
      }
      tabIndex="0"
      viewBox="0 0 15 18"
    >
      <path
        d="M11.5,-0.0004 L0.5,-0.0004 C0.224,-0.0004 0,0.2236 0,0.4996 L0,16.4996 C0,16.6996 0.119,16.8806 0.302, 16.9586 C0.488,17.0396 0.699,16.9996 0.844,16.8636 L6,11.9876 L11.156,16.8636 C11.251,16.9526 11.375, 16.9996 11.5,16.9996 C11.566,16.9996 11.634,16.9866 11.698,16.9586 C11.881,16.8806 12,16.6996 12, 16.4996 L12,0.4996 C12,0.2236 11.776,-0.0004 11.5,-0.0004 L11.5,-0.0004 Z"
        fill="#FFF"
      />
    </svg>
  </div>
</div>
`;

exports[`should render onMouseEnter 2`] = `
<div>
  <div
    aria-label="Bookmarked"
    className="filled"
  >
    <svg
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fill": "currentColor",
          "height": "20px",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "width": "16px",
        }
      }
      tabIndex="0"
      viewBox="0 0 15 18"
    >
      <path
        d="M11.5,-0.0004 L0.5,-0.0004 C0.224,-0.0004 0,0.2236 0,0.4996 L0,16.4996 C0,16.6996 0.119,16.8806 0.302, 16.9586 C0.488,17.0396 0.699,16.9996 0.844,16.8636 L6,11.9876 L11.156,16.8636 C11.251,16.9526 11.375, 16.9996 11.5,16.9996 C11.566,16.9996 11.634,16.9866 11.698,16.9586 C11.881,16.8806 12,16.6996 12, 16.4996 L12,0.4996 C12,0.2236 11.776,-0.0004 11.5,-0.0004 L11.5,-0.0004 Z"
        fill="#FFF"
      />
    </svg>
  </div>
</div>
`;

exports[`should render onMouseLeave 1`] = `
<div>
  <div
    aria-label="Bookmarked"
    className="filled"
  >
    <svg
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fill": "currentColor",
          "height": "20px",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "width": "16px",
        }
      }
      tabIndex="0"
      viewBox="0 0 15 18"
    >
      <path
        d="M11.5,-0.0004 L0.5,-0.0004 C0.224,-0.0004 0,0.2236 0,0.4996 L0,16.4996 C0,16.6996 0.119,16.8806 0.302, 16.9586 C0.488,17.0396 0.699,16.9996 0.844,16.8636 L6,11.9876 L11.156,16.8636 C11.251,16.9526 11.375, 16.9996 11.5,16.9996 C11.566,16.9996 11.634,16.9866 11.698,16.9586 C11.881,16.8806 12,16.6996 12, 16.4996 L12,0.4996 C12,0.2236 11.776,-0.0004 11.5,-0.0004 L11.5,-0.0004 Z"
        fill="#FFF"
      />
    </svg>
  </div>
</div>
`;

exports[`should render onMouseLeave 2`] = `
<div>
  <div
    aria-label="Bookmarked"
    className="filled"
  >
    <svg
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fill": "currentColor",
          "height": "20px",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "width": "16px",
        }
      }
      tabIndex="0"
      viewBox="0 0 15 18"
    >
      <path
        d="M11.5,-0.0004 L0.5,-0.0004 C0.224,-0.0004 0,0.2236 0,0.4996 L0,16.4996 C0,16.6996 0.119,16.8806 0.302, 16.9586 C0.488,17.0396 0.699,16.9996 0.844,16.8636 L6,11.9876 L11.156,16.8636 C11.251,16.9526 11.375, 16.9996 11.5,16.9996 C11.566,16.9996 11.634,16.9866 11.698,16.9586 C11.881,16.8806 12,16.6996 12, 16.4996 L12,0.4996 C12,0.2236 11.776,-0.0004 11.5,-0.0004 L11.5,-0.0004 Z"
        fill="#FFF"
      />
    </svg>
  </div>
</div>
`;
